packages/react-is/src/ReactIs.js

```diff
@@ -116,7 +123,7 @@ export function isElement(object: any) {
   return (
     typeof object === 'object' &&
     object !== null &&
-    object.$$typeof === REACT_ELEMENT_TYPE
+    safelyAccessProperty(object, '$$typeof') === REACT_ELEMENT_TYPE
```

Comment:
I don't understand why we only check this prop within a guard for `isElement` but not for `typeOf`. Why is it okay for one to throw and the other not to?